src/api/query.rs

```diff
@@ -90,13 +109,28 @@ pub fn block_hash_from_string(hash: &str) -> ApiResult<BlockHash> {
     })
 }
 
-pub fn account_address_from_identifier(id: &AccountIdentifier) -> ApiResult<AccountAddress> {
+pub enum Address {
+    Account(AccountAddress),
+    BakingRewardAccount,
+    FinalizationRewardAccount,
+}
+
+pub fn account_address_from_identifier(id: &AccountIdentifier) -> ApiResult<Address> {
     match id.sub_account {
         None => account_address_from_string(&id.address),
         Some(_) => Err(ApiError::SubAccountNotImplemented),
     }
 }
 
-pub fn account_address_from_string(addr: &str) -> ApiResult<AccountAddress> {
-    AccountAddress::from_str(addr).map_err(|_| ApiError::InvalidAccountAddress(addr.to_string()))
+pub fn account_address_from_string(addr: &str) -> ApiResult<Address> {
+    if addr == ACCOUNT_BAKING_REWARD {
+        return Ok(Address::BakingRewardAccount);
+    }
+    if addr == ACCOUNT_FINALIZATION_REWARD {
+        return Ok(Address::FinalizationRewardAccount);
+    }
+    match AccountAddress::from_str(addr) {
+        Ok(a) => Ok(Address::Account(a)),
+        Err(_) => Err(ApiError::InvalidAccountAddress(addr.to_string())),
+    }
```

Comment:
```suggestion
    match addr {
        ACCOUNT_BAKING_REWARD => Ok(Address::BakingRewardAccount),
        ACCOUNT_FINALIZATION_REWARD => Ok(Address::FinalizationRewardAccount),
        _ => {
            match AccountAddress::from_str(addr) {
                Ok(a) => Ok(Address::Account(a)),
                Err(_) => Err(ApiError::InvalidAccountAddress(addr.to_string())),
            }
        }
    }
```
is maybe a bit nicer? :shrug: 
I tried to only have a single match, but it doesn't seem possible without doing stupid stuff

Reply:
Strange, tried that at first but were certain that it interpreted the constants as an ordinary "anything" match. Applied the change manually.